File: czech-railway-transportation/src/main/java/cz/jalasoft/transportation/czechrailway/page/MultipleTrainsPage.java
package cz.jalasoft.transportation.czechrailway.page;

import cz.jalasoft.transportation.czechrailway.page.extraction.ValueExtractors;
import cz.jalasoft.util.text.TextFragment;

import java.util.Collection;

/**
 * Created by honzales on 1.7.15.
 */
public final class MultipleTrainsPage extends Page {

    MultipleTrainsPage(TextFragment content) {
        super(content);
    }

    public Collection<String> extractTrainDetailLinks() {
        return extract(ValueExtractors.trainsDetailsLinks());
    }
}

File: czech-railway-transportation/src/main/scala/cz/jalasoft/transportation/czechrailway/page/MultipleTrainsPage.scala
package cz.jalasoft.transportation.czechrailway.page

import cz.jalasoft.util.text.FragmentConverters._
import cz.jalasoft.util.text.TextFragment

import scala.collection.JavaConversions._

/**
 * Created by honzales on 8.7.15.
 */
final class MultipleTrainsPage (fragment : TextFragment) extends Page(fragment) {

  private val TRAIN_LINK_PATTERN = "<a href=\"(.*)\\?(.*)\" class=\"train thin\""
  //private val TRAIN_LINK_PATTERN = """<a href="(.*)\\?(.*)" class="train thin""""

  def trainsHostsAndPaths : Seq[(String, String)] = {
    val findings = fragment.findFragmentsMatching(TRAIN_LINK_PATTERN)

    val iterable = for(occurrence <- findings) yield {
      val path = occurrence.getGroupTextFragment(0).text
      val params = occurrence.getGroupTextFragment(1).convert(toHtmlEscaped).text

      (path, params)
     }
     iterable.toSeq
  }
  
}
